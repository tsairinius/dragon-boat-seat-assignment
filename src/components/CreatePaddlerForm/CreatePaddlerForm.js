import React, { useState } from "react";
import { v4 as uuidv4 } from "uuid";
import PropTypes from "prop-types";

function CreatePaddlerForm(props) {
  const [paddlerName, setPaddlerName] = useState("");
  const [paddlerGender, setPaddlerGender] = useState("");
  const [paddlerWeight, setPaddlerWeight] = useState("");

  const handleNameChange = (event) => setPaddlerName(event.target.value);
  const handleGenderChange = (event) => setPaddlerGender(event.target.value);
  const handleWeightChange = (event) => setPaddlerWeight(event.target.value);

  const handleSubmit = (event) => {
    event.preventDefault();
    const paddlerProfile = {
      id: uuidv4(),
      name: paddlerName,
      gender: paddlerGender,
      weight: paddlerWeight,
      inBoat: false,
      seatId: "",
      isActive: false,
      isHovered: false,
    };

    props.addPaddler(paddlerProfile);

    handleReset();
  };

  const handleReset = () => {
    setPaddlerName("");
    setPaddlerGender("");
    setPaddlerWeight("");
  };

  return (
    <div style={{ border: "solid 1px green" }}>
      <h1 style={{ color: "black", fontFamily: "Allura", fontSize: "40px" }}>
        Create a Paddler
      </h1>
      <form onSubmit={handleSubmit} onReset={handleReset}>
        <div>
          <label>
            Name
            <input
              type="text"
              name="paddlerName"
              placeholder="e.g. Ed Cheung"
              value={paddlerName}
              onChange={handleNameChange}
              required
            />
          </label>
        </div>
        <div>
          Gender
          <div>
            <label>
              Male
              <input
                type="radio"
                name="paddlerGender"
                value="male"
                checked={paddlerGender === "male"}
                onChange={handleGenderChange}
                required
              />
            </label>
          </div>
          <div>
            <label>
              Female
              <input
                type="radio"
                name="paddlerGender"
                value="female"
                checked={paddlerGender === "female"}
                onChange={handleGenderChange}
                required
              />
            </label>
          </div>
          <div>
            <label>
              Other
              <input
                type="radio"
                name="paddlerGender"
                value="other"
                checked={paddlerGender === "other"}
                onChange={handleGenderChange}
                required
              />
            </label>
          </div>
        </div>
        <div>
          <label>
            Weight(lb)
            <input
              type="number"
              name="paddlerWeight"
              min="1"
              placeholder="e.g. 150"
              value={paddlerWeight}
              onChange={handleWeightChange}
              required
            />
          </label>
        </div>
        <button type="reset">Reset</button>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
}

CreatePaddlerForm.propTypes = {
  label: PropTypes.string.isRequired,
  addPaddler: PropTypes.func.isRequired,
};

export default CreatePaddlerForm;
