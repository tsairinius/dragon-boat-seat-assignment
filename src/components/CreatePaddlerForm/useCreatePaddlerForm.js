import { useState } from "react";
import { v4 as uuidv4 } from "uuid";

function useCreatePaddlerForm(addPaddler) {
  const [paddlerName, setPaddlerName] = useState("");
  const [paddlerGender, setPaddlerGender] = useState("");
  const [paddlerWeight, setPaddlerWeight] = useState("");

  const handleNameChange = (event) => setPaddlerName(event.target.value);
  const handleGenderChange = (event) => setPaddlerGender(event.target.value);
  const handleWeightChange = (event) => setPaddlerWeight(event.target.value);

  const handleSubmit = (event) => {
    event.preventDefault();
    const paddlerProfile = {
      id: uuidv4(),
      name: paddlerName,
      gender: paddlerGender,
      weight: paddlerWeight,
      inBoat: false,
      seatId: "",
      isActive: false,
      isHovered: false,
    };

    addPaddler(paddlerProfile);

    handleReset();
  };

  const handleReset = () => {
    setPaddlerName("");
    setPaddlerGender("");
    setPaddlerWeight("");
  };

  return {
    paddlerName,
    paddlerGender,
    paddlerWeight,
    handleNameChange,
    handleGenderChange,
    handleWeightChange,
    handleSubmit,
    handleReset,
  };
}

export default useCreatePaddlerForm;
