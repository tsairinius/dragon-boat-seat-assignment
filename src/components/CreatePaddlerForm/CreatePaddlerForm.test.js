import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CreatePaddlerForm from './CreatePaddlerForm';

afterEach(cleanup);

it('renders CreatePaddlerForm component without crashing', () => {
  const container = document.createElement('div');
  ReactDOM.render(<CreatePaddlerForm />, container);
});

it('rendered CreatePaddlerForm matches snapshot', () => {
    const tree = renderer.create(<CreatePaddlerForm />).toJSON();
    expect(tree).toMatchSnapshot();
});

