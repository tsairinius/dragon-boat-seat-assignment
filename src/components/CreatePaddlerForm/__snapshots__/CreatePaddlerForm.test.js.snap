// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered CreatePaddlerForm matches snapshot 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div>
    <label>
      Name
    </label>
    <input
      name="paddlerName"
      onChange={[Function]}
      placeholder="e.g. Ed Cheung"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div>
    Gender
    <div>
      <label>
        Male
      </label>
      <input
        checked={false}
        name="paddlerGender"
        onChange={[Function]}
        required={true}
        type="radio"
        value="male"
      />
    </div>
    <div>
      <label>
        Female
      </label>
      <input
        checked={false}
        name="paddlerGender"
        onChange={[Function]}
        required={true}
        type="radio"
        value="female"
      />
    </div>
    <div>
      <label>
        Other
      </label>
      <input
        checked={false}
        name="paddlerGender"
        onChange={[Function]}
        required={true}
        type="radio"
        value="other"
      />
    </div>
  </div>
  <div>
    <label>
      Weight(lb)
    </label>
    <input
      min="1"
      name="paddlerWeight"
      onChange={[Function]}
      placeholder="e.g. 150"
      required={true}
      type="number"
      value=""
    />
  </div>
  <button
    type="reset"
  >
    Reset
  </button>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
